crates/wallet/src/wallet/mod.rs

```diff
@@ -1456,12 +1456,12 @@ impl Wallet {
                 required_utxos.clone(),
                 optional_utxos.clone(),
                 fee_rate,
-                outgoing.to_sat() + fee_amount.to_sat(),
+                outgoing + fee_amount,
                 &drain_script,
                 rng,
             )
             .map_err(CreateTxError::CoinSelection)?;
-        fee_amount += Amount::from_sat(coin_selection.fee_amount);
+        fee_amount += coin_selection.fee_amount;
```

Comment:
`fee_amount` isn't used after this point (it is only assigned to), so we should remove it.

Reply:
I see `fee_amount` is assigned to after L1464 on L1508, L1513, but as you say it isn't used. The code is pretty convoluted but it seems none of the lines that changes `fee_amount` after L1459 are needed. That also seems to be true for the `received` value on L1511. 

I suspect some of this code was added during development for debugging and never removed. I'd rather not make unrelated changes in this PR, do you mind if I create a new issue to research and clean up the `Wallet::create_tx()` logic?

Reply:
I assumed it was left over from when `create_tx` returned a tuple of `(Psbt, TransactionDetails)`. https://docs.rs/bdk/latest/src/bdk/wallet/mod.rs.html#603

Reply:
Ah ha, I think you're correct and I'm responsible for leaving it there when I did #1048. I'll create a new PR to clean it up, it doesn't affect any APIs so can wait until post 1.0.0.

Reply:
See #1768 .